### PR TITLE
Fix persian's half-space's visibility and add a 0.5 width before it's ظ

### PR DESCRIPTION
**Pull request by RZHSSNZDH**
Related to https://github.com/Julow/Unexpected-Keyboard/pull/709. I tryied to fix the problems.

### CODE REVIEW

**Comment by RZHSSNZDH** [comment]
Please do not merge this PR. I am debugging :)

**Comment by RZHSSNZDH** [comment]
> Please do not merge this PR. I am debugging :)

Finished!

File: srcs/juloo.keyboard2/KeyValue.java [hunk 1]
```diff
@@ -551,6 +551,8 @@ public static KeyValue getKeyByName(String name)
       /* intending/preventing ligature - supported by many scripts*/
       case "zwj": return charKey("zwj", '\u200D', 0); // zero-width joiner (provides ligature)
       case "zwnj": return charKey("zwnj", '\u200C', 0); // zero-width non joiner (prevents unintended ligature)
+      /* persian halfspace */
+      case "halfspace": return charKey("⸽", '\u200C', 0);
```
Comment (Julow) [inline srcs/juloo.keyboard2/KeyValue.java:555]: The new `halfspace` key outputs exactly the same character as `zwnj`. What do you think of using `zwnj` in your layout instead ? You can put the better rendering symbol in `zwnj` instead.

Comment (RZHSSNZDH) [reply srcs/juloo.keyboard2/KeyValue.java:555]: Its logical, but I thought the `zwnj` word maybe is a proper noun in a language which maybe is not correct to rename it.

Comment (RZHSSNZDH) [reply srcs/juloo.keyboard2/KeyValue.java:555]: Yes it is logical, but I thought maybe the word `zwnj` maybe is a proper noun in a language and maybe its not correct to rename it.

Comment (Julow) [reply srcs/juloo.keyboard2/KeyValue.java:555]: `zwnj` is the Unicode name (zero-width non joiner) and is ugly on the keyboard. It's not intended to mean anything in any language. So yes, let's change it for `⸽`.

Comment (RZHSSNZDH) [reply srcs/juloo.keyboard2/KeyValue.java:553]: Ok.
https://github.com/Julow/Unexpected-Keyboard/pull/711/commits/cf96ab4b7abe1fdeadcffca36874a6f7dd365d41

File: srcs/juloo.keyboard2/KeyValue.java [hunk 2]
```diff
@@ -550,7 +550,7 @@ public static KeyValue getKeyByName(String name)
       case "meteg_placeholder": return placeholderKey(Placeholder.METEG);
       /* intending/preventing ligature - supported by many scripts*/
       case "zwj": return charKey("zwj", '\u200D', 0); // zero-width joiner (provides ligature)
-      case "zwnj": return charKey("zwnj", '\u200C', 0); // zero-width non joiner (prevents unintended ligature)
+      case "halfspace": return charKey("⸽", '\u200C', 0);
```
Comment (Julow) [inline srcs/juloo.keyboard2/KeyValue.java:553]: We still need to support layouts that were using `zwnj` until now:

```suggestion
      case "zwnj":
      case "halfspace": return charKey("⸽", '\u200C', 0); // zero-width non joiner
```

**Review by Julow — APPROVED** [review]
Thanks!

**Pull request merged**